eslint-plugin/rules/no-window-outside-useeffect.test.js

```diff
@@ -0,0 +1,103 @@
+const rule = require('./no-window-outside-use-effect');
+const RuleTester = require('eslint').RuleTester;
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-window-outside-use-effect', rule, {
+  valid: [
+    // window used inside useEffect
+    {
+      code: `
+        import { useEffect } from 'react';
+
+        function MyComponent() {
+          useEffect(() => {
+            window.alert('Hello, world!');
+          }, []);
+        
+          return null;
+        }
+      `,
+    },
```

Comment:
```suggestion
    const windowUsedInsideUseEffect = {
      code: `
        import { useEffect } from 'react';

        function MyComponent() {
          useEffect(() => {
            window.alert('Hello, world!');
          }, []);
        
          return null;
        }
      `,
    };
```
And you don't need comments.